src/pages/workspaces/workspace/useDeleteWorkspaceState.ts

```diff
@@ -35,83 +53,156 @@ export interface DeleteWorkspaceHookArgs {
           Ajax(signal).Workspaces.workspace(workspaceInfo.namespace, workspaceInfo.name).bucketUsage()
         ])
-        setApps(currentWorkspaceAppList)
         setCollaboratorEmails(_.without([getUser().email], _.keys(acl)))
         setWorkspaceBucketUsageInBytes(usageInBytes)
-      } else {
-        const currentWorkspaceAppList = await Ajax(signal).Apps.listAppsV2(workspaceInfo.workspaceId)
-
-        // temporary hack to prevent orphaning resources on Azure:
-        // change each app to status: 'disallow' which will cause this modal to think they are undeletable
-        const hackedAppList = _.map(_.set('status', 'disallow'), currentWorkspaceAppList) as DeleteWorkspaceModalLeoApp[]
-        setApps(hackedAppList)
-
-        // Also temporarily disable delete if there are any controlled resources besides the expected workspace storage container.
-        const controlledResources = await Ajax(signal).WorkspaceManagerResources.controlledResources(workspaceInfo.workspaceId)
-        setControlledResourcesExist(controlledResources.resources.length > 1)
       }
     })
     load()
+
+    return () => clearTimeout(checkAzureResourcesTimeout.current)
   })
 
   const hasApps = () => {
-    return deletableApps !== undefined && nonDeletableApps !== undefined &&
-        (!_.isEmpty(deletableApps) ||
-            !_.isEmpty(nonDeletableApps))
+    return (workspaceResources !== undefined && !_.isEmpty(workspaceResources.apps))
   }
 
-  const isDeleteDisabledFromResources = (hasApps() && !_.isEmpty(nonDeletableApps)) || controlledResourcesExist
+  const hasRuntimes = () => {
+    return (workspaceResources !== undefined && !_.isEmpty(workspaceResources.runtimes))
+  }
+
+  const isDeleteDisabledFromResources = workspaceResources !== undefined && (
+    (hasApps() && !_.isEmpty(workspaceResources.nonDeleteableApps)) || (hasRuntimes() && !_.isEmpty(workspaceResources.nonDeleteableRuntimes))
+  )
 
   const deleteWorkspace = async () => {
+    if (isDeleteDisabledFromResources) {
+      throw new Error('Workspace contains non-deletable resources')
+    }
+
     try {
       setDeleting(true)
-      if (isGoogleWorkspace(hookArgs.workspace)) {
+      if (isGoogleWorkspace(hookArgs.workspace) && workspaceResources) {
         await Promise.all(
-          _.map(async app => await Ajax().Apps.app(app.cloudContext.cloudResource, app.appName).delete(), deletableApps)
+          _.map(async app => await Ajax(signal).Apps.app(app.cloudContext.cloudResource, app.appName).delete(), workspaceResources.deleteableApps)
         )
       }
-      await Ajax().Workspaces.workspace(workspaceInfo.namespace, workspaceInfo.name).delete()
+
+      await Ajax(signal).Workspaces.workspace(workspaceInfo.namespace, workspaceInfo.name).delete()
       hookArgs.onDismiss()
       hookArgs.onSuccess()
     } catch (error) {
-      reportError('Error deleting workspace', error)
       setDeleting(false)
+      await reportError('Error deleting workspace', error)
```

Comment:
I don't think you need to await `reportError`, as this is the end of the method and you don't care when the display of the error has been done. Or did you add the await specifically for testing purposes?

Reply:
No reason other than to get rid of a yellow ide squiggle. Will remove.